netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -203,7 +204,14 @@ public void deframeFailed(Throwable cause) {
       log.log(Level.WARNING, "Exception processing message", cause);
       Status status = Status.fromThrowable(cause);
       transportReportStatus(status);
-      handler.getWriteQueue().enqueue(new CancelServerStreamCommand(this, status), true);
+
+      Metadata trailers = new Metadata();
+      trailers.put(InternalStatus.CODE_KEY, status);
+      if (status.getDescription() != null) {
+        trailers.put(InternalStatus.MESSAGE_KEY, status.getDescription());
+      }
+      Http2Headers http2Trailers = Utils.convertTrailers(trailers, /* headersSent = */ false);
+      handler.getWriteQueue().enqueue(SendResponseHeadersCommand.createTrailers(this, http2Trailers, status), /* flush = */ true);
```

Comment:
this is the meat of the change — rather than simply canceling the stream, close it out gracefully by sending the proper trailers